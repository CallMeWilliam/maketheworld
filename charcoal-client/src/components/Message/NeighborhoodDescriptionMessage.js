import React from 'react'

import {
    Typography,
    ListItem,
    ListItemText,
    ListItemIcon
} from '@material-ui/core'
import CityIcon from '@material-ui/icons/LocationCity'

import useStyles from '../styles'

export const NeighborhoodDescriptionMessage = ({ message, ...rest }) => {

    const classes = useStyles()
    const { Name='', Description='' } = message || {}

    return <ListItem className={ classes.neighborhoodMessage } alignItems="flex-start" {...rest} >
            <ListItemIcon>
                <CityIcon />
            </ListItemIcon>
            <ListItemText>
                <Typography variant='h5' align='left'>
                    { Name }
                </Typography>
                <Typography variant='body1' align='left'>
                    { Description }
                </Typography>
            </ListItemText>
        </ListItem>
}

export default NeighborhoodDescriptionMessage