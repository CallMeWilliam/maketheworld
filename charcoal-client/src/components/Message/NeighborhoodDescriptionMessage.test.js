import React from 'react';
import { render } from '@testing-library/react';
import NeighborhoodDescriptionMessage from './NeighborhoodDescriptionMessage';

describe('NeighborhoodDescriptionMessage component', () => {
    it('should render empty message', () => {
        const { container } = render(<NeighborhoodDescriptionMessage />)
        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render test message', () => {
        const { container } = render(<NeighborhoodDescriptionMessage message={{
                Description: "This is a test",
                Name: "Test"
            }}/>)
        expect(container.firstChild).toMatchSnapshot()
    })
})

