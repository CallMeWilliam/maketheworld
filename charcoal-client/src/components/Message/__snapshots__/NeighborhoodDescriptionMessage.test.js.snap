// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeighborhoodDescriptionMessage component should render empty message 1`] = `
<li
  class="MuiListItem-root makeStyles-roomMessage-25 MuiListItem-gutters MuiListItem-alignItemsFlexStart"
>
  <div
    class="MuiListItemIcon-root MuiListItemIcon-alignItemsFlexStart"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 9.3V4h-3v2.6L12 3 2 12h3v8h5v-6h4v6h5v-8h3l-3-2.7zm-9 .7c0-1.1.9-2 2-2s2 .9 2 2h-4z"
      />
    </svg>
  </div>
  <div
    class="MuiListItemText-root"
  >
    <span
      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft"
      />
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft"
      />
    </span>
  </div>
</li>
`;

exports[`NeighborhoodDescriptionMessage component should render test message 1`] = `
<li
  class="MuiListItem-root makeStyles-roomMessage-128 MuiListItem-gutters MuiListItem-alignItemsFlexStart"
>
  <div
    class="MuiListItemIcon-root MuiListItemIcon-alignItemsFlexStart"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 9.3V4h-3v2.6L12 3 2 12h3v8h5v-6h4v6h5v-8h3l-3-2.7zm-9 .7c0-1.1.9-2 2-2s2 .9 2 2h-4z"
      />
    </svg>
  </div>
  <div
    class="MuiListItemText-root"
  >
    <span
      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft"
      >
        Test
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft"
      >
        This is a test
      </p>
    </span>
  </div>
</li>
`;
